### PR TITLE
[casacore] Update to 3.5.0

### PR DESCRIPTION
**Pull request by kiranshila**
They merged the two patches we included

### CODE REVIEW

**Comment by giordano** [comment]
They require python now?

**Comment by kiranshila** [comment]
I sure hope not, looks like they changed a few deps though

**Comment by kiranshila** [comment]
Ah they added a new flag, `BUILD_PYTHON3`

File: C/casacore/build_tarballs.jl [hunk 1]
```diff
@@ -112,6 +108,8 @@ dependencies = [
     Dependency(PackageSpec(name="CFITSIO_jll", uuid="b3e40c51-02ae-5482-8a39-3ace5868dcf4"); compat = "4.0.0")
     Dependency(PackageSpec(name="WCS_jll", uuid="550c8279-ae0e-5d1b-948f-937f2608a23e"); compat = "7.7.0")
     Dependency(PackageSpec(name="Readline_jll", uuid="05236dd9-4125-5232-aa7c-9ec0c9b2c25a"); compat = "8.1.1")
+    Dependency(PackageSpec(name="boost_jll", uuid="28df3c45-c428-5900-9ff8-a3135698ca75"); compat = "1.76.0")
```
Comment (giordano) [inline C/casacore/build_tarballs.jl:111]: Is boost really necessary?  It's a pain, it constantly breaks ABI: #3024.  If you want to use it, you have to pin it:
```suggestion
    Dependency(PackageSpec(name="boost_jll", uuid="28df3c45-c428-5900-9ff8-a3135698ca75"); compat = "=1.76.0")
```

Comment (giordano) [reply C/casacore/build_tarballs.jl:111]: But according to https://github.com/casacore/casacore/blob/ff766c49fdfddd34715fd017a08aaedcc01736e0/README.md#requirements boost isn't required.

Comment (kiranshila) [reply C/casacore/build_tarballs.jl:111]: Ah, the error was a red herring for GSL.

**Pull request merged**